### PR TITLE
Issue 222: remove features

### PR DESCRIPTION
**Pull request by LukasOsti**
Implemented the removing of features via Slicing only available with the feature IDE service running.

### CODE REVIEW

File: frontendVue3/src/views/FeatureModel.vue [hunk 1]
```diff
@@ -178,6 +193,56 @@ export default {
...
+            if (this.isServiceAvailable) {
+
+                this.xml = jsonToXML(this.data);
+                const content = new TextEncoder().encode(this.xml);
+                let response = await axios.post(`${import.meta.env.VITE_APP_DOMAIN_FEATUREIDESERVICE}slice`, {
+                    name: "hello.xml",
+                    selection: [node.name],
+                    content: Array.from(content)
+                });
+                console.log(response);
```
Comment (tomatenhans) [inline frontendVue3/src/views/FeatureModel.vue]: Maybe consider removing console logs.

File: frontendVue3/src/views/FeatureModel.vue [hunk 2]
```diff
@@ -178,6 +193,56 @@ export default {
...
+                let contentAsString = new TextDecoder().decode(Uint8Array.from(response.data.content));
+                console.log(contentAsString)
```
Comment (tomatenhans) [inline frontendVue3/src/views/FeatureModel.vue]: Maybe consider removing console logs.

File: frontendVue3/src/views/FeatureModel.vue [hunk 3]
```diff
@@ -178,6 +193,56 @@ export default {
             this.reloadKey++;
         },
 
+        async slice(node) {
+            this.loadingData = true;
+            console.log(this.loadingData)
```
Comment (tomatenhans) [inline frontendVue3/src/views/FeatureModel.vue]: Maybe consider removing console logs.

**Review by tomatenhans — CHANGES_REQUESTED** [review]
Code looks fine and works fine on my PC :)
As mentioned maybe remove the console logs. 

Only odd thing is the insertion of new Root features when removing stuff, but as discussed this is a featureIDE issue.

**Review by MeisterSeSe — COMMENTED** [review]
image
Is it intended that features which are direct children of the root cannot be removed? The code appears to be fine apart from the console logs that were already mentioned.

**Comment by LukasOsti** [comment]
It was intended in the first place, but i changed the code to allow more trivial removing without the service. So your example should work now without the service.

**Pull request merged**